### PR TITLE
[en/Scala] Rewrite the pattern matching section

### PR DESCRIPTION
**Pull request by geoffliu**

### CODE REVIEW

File: scala.html.markdown [hunk 1]
```diff
@@ -407,41 +407,55 @@ val otherGeorge = george.copy(phoneNumber = "9876")
 // 6. Pattern Matching
 /////////////////////////////////////////////////
 
-val me = Person("George", "1234")
+// Pattern matching is a powerful and commonly used feature in Scala. Here's how
+// you pattern match a case class. NB: Unlike other languages, Scala cases do
+// not have breaks.
```
Comment (levibostian) [inline scala.html.markdown]: If Scala doesn't have breaks, what do you do instead?

Comment (geoffliu) [inline scala.html.markdown]: Ah, poor wording. Scala doesn't need breaks between cases, fall-through doesn't happen. I'll update.

**Comment by levibostian** [comment]
Thanks!

File: scala.html.markdown (hunk 1 above)
Comment (levibostian) [inline scala.html.markdown]: Ok thanks. Go ahead and make another PR when you have it complete.
